### PR TITLE
Add metadata files in ingest

### PR DESCRIPTION
**Pull request by Hwesta**
Added the ability to upload metadata files from a transfer source directory to a SIP in ingest.

From the metadata page in Ingest, added a link to "Add metadata files". Here, you can browse transfer source directories to select files to upload to the SIP's objects/metadata directory. The preferred place to do this is the add metadata wait point, which was moved to before metadata & submission docs are processed.  Converting a metadata.json to a metadata.csv was also added to process metadata files.

Cleaned up the code that parses metadata.csv into file level DC.

metadata.csv files can be added in Transfer and in Ingest. If information conflicts, the Ingest metadata will completely overwrite anything found in transfer. There is also a known bug that if the columns (or column order) differ between the transfer and ingest metadata.csvs, data may end up in the wrong DC tag.

Related to AIP Reingest

### CODE REVIEW

File: src/MCPClient/lib/clientScripts/archivematicaCreateMETSMetadataCSV.py [hunk 1]
```diff
@@ -51,16 +52,23 @@ def parseMetadata(SIPPath):
...
             try:
-                parseMetadtaCSV(metadataCSVFilePath)
-            except Exception as inst:
-                print >>sys.stderr, type(inst)     # the exception instance
-                print >>sys.stderr, inst.args
+                parseMetadataCSV(metadataCSVFilePath)
+            except Exception:
                 print >>sys.stderr, "error parsing: ", metadataCSVFilePath
-                traceback.print_exc(file=sys.stdout)
+                traceback.print_exc(file=sys.stderr)
```
Comment (mistydemeo) [inline src/MCPClient/lib/clientScripts/archivematicaCreateMETSMetadataCSV.py:58]: :thumbsup:

**Comment by mistydemeo** [comment]
> metadata.csv files can be added in Transfer and in Ingest. If information conflicts, the Ingest metadata will completely overwrite anything found in transfer.

Let's make sure to log that condition, so there's a debug trail for Archivematica operators who unexpectedly redefine their metadata.

**Comment by Hwesta** [comment]
> > metadata.csv files can be added in Transfer and in Ingest. If information conflicts, the Ingest metadata will completely overwrite anything found in transfer.
> 
> Let's make sure to log that condition, so there's a debug trail for Archivematica operators who unexpectedly redefine their metadata.

Not sure that's easily doable currently... I'll look into it.

**Comment by mistydemeo** [comment]
Are we planning this for 1.4.0?

File: src/MCPClient/lib/clientScripts/archivematicaCreateMETS2.py [hunk 2]
```diff
@@ -153,18 +155,28 @@ def getDublinCore(unit, id):
...
+        # Restructure data
+        # Create OrderedDict with keys from compoundMetadataCSVkey and values for directory
+        # Drop first element, since that is the directory name
+        metadata = collections.OrderedDict(zip(compoundMetadataCSVkey[1:], compoundMetadataCSV[directory][1:]))
+        dmdsecs = createDmdSecsFromCSVParsedMetadata(metadata)
+        return ' '.join([d.get('ID') for d in dmdsecs])
 
 
-def createDMDIDSFromCSVParsedMetadataPart2(keys, values):
+def createDmdSecsFromCSVParsedMetadata(metadata):
```
Comment (mistydemeo) [inline src/MCPClient/lib/clientScripts/archivematicaCreateMETS2.py:179]: :+1: on new title

File: src/MCPClient/lib/clientScripts/archivematicaCreateMETSMetadataCSV.py [hunk 3]
```diff
@@ -51,28 +52,36 @@ def parseMetadata(SIPPath):
...
+        try:
+            parseMetadataCSV(metadataCSVFilePath)
+        except Exception:
+            print >>sys.stderr, "error parsing: ", metadataCSVFilePath
+            traceback.print_exc(file=sys.stderr)
+            sharedVariablesAcrossModules.globalErrorCount += 1
 
 
-def parseMetadtaCSV(metadataCSVFilePath):
+def parseMetadataCSV(metadataCSVFilePath):
```
Comment (mistydemeo) [inline src/MCPClient/lib/clientScripts/archivematicaCreateMETSMetadataCSV.py:71]: Also :+1: on fixing this title

**Comment by mistydemeo** [comment]
:+1:

**Pull request merged**